src/test/javascript/spec/service/guided-tour.service.spec.ts

```diff
@@ -21,149 +21,206 @@ import { GuidedTourComponent } from 'app/guided-tour/guided-tour.component';
+            it('should prevent backdrop from advancing', () => {
+                const backdrop = guidedTourComponentFixture.debugElement.queryAll(By.css('.guided-tour-overlay'));
+                expect(backdrop).to.exist;
+                expect(backdrop.length).to.equal(4);
+                backdrop.forEach(overlay => {
+                    overlay.nativeElement.click();
                 });
+                guidedTourComponentFixture.detectChanges();
+                expect(guidedTourService.isOnFirstStep).to.be.true;
+            });
+        });
 
-                it('should start and finish the course overview guided tour', () => {
-                    // Navigate to next tour step
-                    const nextButton = guidedTourComponentFixture.debugElement.query(By.css('.next-button'));
-                    expect(nextButton).to.exist;
-                    nextButton.nativeElement.click();
-                    expect(guidedTourService.isOnLastStep).to.be.true;
+        describe('Tours with user interaction', () => {
+            beforeEach(async () => {
+                await prepareGuidedTour(courseOverviewTourWithUserInteraction);
+            });
 
-                    // Finish guided tour
-                    nextButton.nativeElement.click();
-                    guidedTourComponentFixture.detectChanges();
-                    expect(guidedTourComponentFixture.debugElement.query(By.css('.tour-step'))).to.not.exist;
-                });
+            it('should enable user interaction and navigate to next step after user interaction', async () => {
+                // Fix jsom error: Not implemented: window.scrollTo
+                const jsdomAlert = window.scrollTo;
+                window.scrollTo = () => {};
 
-                it('should start and skip the tour', () => {
-                    const skipButton = guidedTourComponentFixture.debugElement.query(By.css('.close'));
-                    expect(skipButton).to.exist;
-                    skipButton.nativeElement.click();
-                    guidedTourComponentFixture.detectChanges();
-                    expect(guidedTourComponentFixture.debugElement.query(By.css('.tour-step'))).to.not.exist;
-                });
+                const selector = navbarComponentFixture.debugElement.query(By.css('.random-selector'));
 
-                it('should prevent backdrop from advancing', () => {
-                    const backdrop = guidedTourComponentFixture.debugElement.query(By.css('.guided-tour-user-input-mask'));
-                    expect(backdrop).to.exist;
-                    backdrop.nativeElement.click();
-                    guidedTourComponentFixture.detectChanges();
-                    expect(guidedTourService.isOnFirstStep).to.be.true;
-                });
+                if (selector) {
+                    selector.nativeElement.click();
+                    await new Promise(resolve => setTimeout(() => resolve(), 500));
```

Comment:
Sorry, this came to me later after the review: It would be better use `fakeAsync` with `tick` here. Otherwise the test actually waits 500 ms here.